web/libs/frontend-test/src/helpers/LSF/AudioView.ts

```diff
@@ -334,6 +348,325 @@ class AudioViewHelper extends withMedia(
+
+          // Return the last sampled color
+          return color;
+        });
+      };
+
+      return sampleColor();
+    }
+
+    /**
+     * Compares two color arrays for equality
+     * @param color1 first color array
+     * @param color2 second color array
+     */
+    colorsEqual(color1: Uint8ClampedArray, color2: Uint8ClampedArray): boolean {
+      if (color1.length !== color2.length) return false;
+      for (let i = 0; i < color1.length; i++) {
+        if (color1[i] !== color2[i]) return false;
+      }
+      return true;
+    }
+
+    /**
+     * Waits for canvas rendering to stabilize by checking that pixel colors remain consistent
+     * @param x relative x coordinate to monitor
+     * @param y relative y coordinate to monitor
+     * @param stabilityChecks number of consecutive stable checks required
+     * @param timeout maximum time to wait in milliseconds
+     */
+    waitForCanvasStable(x = 0.36, y = 0.9, stabilityChecks = 3, timeout = 10000) {
+      let stableCount = 0;
+      let lastColor: Uint8ClampedArray | null = null;
+      const startTime = Date.now();
+
+      const checkStability = (): Cypress.Chainable => {
+        if (Date.now() - startTime > timeout) {
+          cy.log(`⏰ Canvas stabilization timeout after ${timeout}ms`);
+          return cy.wrap(null);
+        }
+
+        return this.getPixelColorRelative(x, y).then((currentColor) => {
+          // Check if we're getting transparent pixels (indicates canvas not ready)
+          const isTransparent =
+            currentColor[0] === 0 && currentColor[1] === 0 && currentColor[2] === 0 && currentColor[3] === 0;
+
+          if (isTransparent) {
+            cy.log("🔍 Canvas not ready - transparent pixel detected, continuing to wait...");
+            stableCount = 0;
+            lastColor = null;
+            cy.wait(100); // Wait longer for CI environments
+            return checkStability();
+          }
+
+          if (lastColor && this.colorsEqual(lastColor, currentColor)) {
+            stableCount++;
+            cy.log(`✅ Canvas stable check ${stableCount}/${stabilityChecks}`);
+            if (stableCount >= stabilityChecks) {
+              cy.log(`🎯 Canvas stabilized after ${stableCount} consecutive checks`);
+              return cy.wrap(null);
+            }
+          } else {
+            stableCount = 0;
+            cy.log("🔄 Canvas changed, resetting stability counter");
+          }
+
+          lastColor = currentColor;
+          cy.wait(16); // Wait one frame
+          return checkStability();
+        });
+      };
+
+      cy.log(`🏁 Starting canvas stabilization check at (${x}, ${y})`);
+      return checkStability();
+    }
+
+    /**
+     * Waits for canvas to have actual content (non-transparent pixels)
+     * @param timeout maximum time to wait in milliseconds
+     */
+    waitForCanvasContent(timeout = 15000) {
+      const startTime = Date.now();
+
+      const checkForContent = (): Cypress.Chainable => {
```

Comment:
Most of the checks like that could be done using cypress built-in features with internal timeouts and retries. https://github.com/HumanSignal/label-studio/blob/develop/web/libs/frontend-test/src/helpers/LSF/VideoView.ts#L371-L385